### PR TITLE
Easier tests

### LINKED ISSUE #109: Easier test running
In https://github.com/sportngin/okcomputer/pull/96 I removed the default fallback of choosing the most recent supported version of Rails when running tests. Instead, you must specify a Rails version to test against when running tests.

This was intended to make CI fail harder when it's given a version of Rails in the build matrix it doesn't recognize, but has the unintended consequence of making running local tests slightly more cumbersome during development.

I'd like it if tests were easier to run for humans (by defaulting to the highest supported version of Rails, able to be overwritten by the environment variable) while still requiring explicit declaration in Travis.

### PR DESCRIPTION
**Pull request by chrisarcand**
Fixes #109 

* Specs will be run locally with `rake spec`
* Specs will default locally to Rails 5.0, minimizing newcomer confusion.
* Removes ugly case statement in Gemfile
* Also removes apparent old Appraisal gem lockfiles. I'm not sure if that's something we care to *actually* use here or not (@pbyrne? Looks like you've seen this gem before), my own opinion is not to bother as reresolving your lockfile just to switch Rails versions seems perfectly simplistic to me.

### CODE REVIEW

**Comment by pbyrne** [comment]
This looks good. At some point I must have been trying to set things up with [`appraisal` to manage testing against multiple dependencies](https://github.com/thoughtbot/appraisal) but it seems I never pulled the trigger (beyond copying some, but not all, configuration for it). Deleting those related files seems 💯 since Travis handles that for us automatically (along with multiple Ruby versions).

File: Rakefile [hunk 1]
```diff
@@ -9,13 +9,23 @@ Bundler::GemHelper.install_tasks
...
 task :default => :spec
 
-require "rspec/core/rake_task"
+begin
+  require "rspec/core/rake_task"
 
-RSpec::Core::RakeTask.new(:spec)
+  RSpec::Core::RakeTask.new(:spec) do |t|
+    if ENV["CI"]
+      raise "RAILS_VERSION must be specified on CI" unless ENV["RAILS_VERSION"]
```
Comment (pbyrne) [inline Rakefile]: This makes me a little sad, but I don't have a good alternative to suggest.

Comment (chrisarcand) [inline Rakefile]: ¯\\\_(ツ)_/¯ Why? you wanted explicit vars set for CI. We could just remove it and output on CI to be easier to see when things actually do go wrong.

Comment (pbyrne) [inline Rakefile]: > Why?

A few fuzzy reasons, mostly.

* It decouples it from the check against `RAILS_VERSION` in the Gemfile where the failure happens now.
* Which also gives us two places to change if we ever change this mechanism.
* Raw `raise`s always make me sad.

So thinking harder about it, I think I have a counterproposal…. Maybe something like this in the Gemfile case statement:

```ruby
case ENV['RAILS_VERSION']
  # all the normal checks we have now
  else
    gem "rails", "~> 5.0.0" unless ENV["CI"]
end
```

Pretty sure I don't like that either, but it at least keeps all the Rails version stuff in one place.

Comment (chrisarcand) [inline Rakefile]: Yup, I think I like that better.

**Comment by pbyrne** [comment]
> CI still requires `RAILS_VERSION` to be explicitly set.

I'm beginning to question how relevant/necessary that requirement is. It was borne out of a mistake I made when upgrading to Rails 5. I added it to the build matrix and everything was green. I only later discovered that it was green because it was falling back to Rails 4.2 because the Gemfile didn't know what the hell I was talking about (we used an `else` in the case statement against `ENV['RAILS_VERSION']`).

So… maybe there's a better way? Maybe we don't even _need_ the case statement in the Gemfile? What if it looked like this instead?

```ruby
source "http://rubygems.org"
RAILS_VERSION = ENV.fetch("RAILS_VERSION", "5.0")

gem "rails", "~> #{RAILS_VERSION}.0"
gem 'test-unit', '~> 3.0' if RUBY_VERSION >= "2.2" && RAILS_VERSION == "3.2"

gemspec
```

We'd still default to a reasonable version of Rails and the Travis build matrix will still load the right version of Rails. I think.

**Comment by chrisarcand** [comment]
Alrighty, I'm done tinkering here.

**Comment by pbyrne** [comment]
I haven't done the due diligence to verify in the Travis output that it's actually loading the right versions of Rails in each build. If you've done that, I say squash and ship this.

**Comment by anfleene** [comment]
> I haven't done the due diligence to verify in the Travis output that it's actually loading the right versions of Rails in each build
I did. @chrisarcand :shipit:

**Pull request merged**